revisions/announcer/announcer_test.go

```diff
@@ -0,0 +1,606 @@
+// +build small
+
+// Copyright 2018 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package announcer_test
+
+import (
+	"errors"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/web-platform-tests/wpt.fyi/revisions/announcer"
+	"github.com/web-platform-tests/wpt.fyi/revisions/epoch"
+	agit "github.com/web-platform-tests/wpt.fyi/revisions/git"
+	"github.com/web-platform-tests/wpt.fyi/revisions/test"
+	billy "gopkg.in/src-d/go-billy.v4"
+	git "gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
+	"gopkg.in/src-d/go-git.v4/plumbing/storer"
+	"gopkg.in/src-d/go-git.v4/storage"
+)
+
+type Fake struct{}
+
+var errFake = errors.New("Implementation is fake")
```

Comment:
You're better off using [`mockgen`](https://github.com/golang/mock) here, since that throws when unexpected calls occur and the tests can more easily see what's being (mocked) returned.
See https://github.com/web-platform-tests/wpt.fyi/blob/master/api/receiver/results_receiver_test.go for a recent example in our codebase.

Reply:
I see invocations of `NewMockAppEngineAPI`, but no implementation.

Reply:
Done.